test.sh

```diff
@@ -24,3 +26,19 @@ yarn test
 
 # Run end-to-end tests
 # TODO
+
+# Upload code coverage
+if [[ "${JOB_TYPE}" == "presubmit" ]]; then
+       echo "detected PR code coverage job for #${PULL_NUMBER}"
+       REF_FLAGS="-P ${PULL_NUMBER} -C ${PULL_PULL_SHA}"
+elif [[ "${JOB_TYPE}" == "postsubmit" || "${JOB_TYPE}" == "periodic" ]]; then
+       REF_FLAGS=""
+else
+       echo "Coverage not enabled on Job Type :${JOB_TYPE}"
+fi
+
+if [[ "${JOB_TYPE}" != "local" ]]; then
+  curl -Os https://uploader.codecov.io/latest/linux/codecov
```

Comment:
Will this fail the job if we fail to upload the codecov?

Reply:
I don't think so, the token was missing previously and it just didn't published the report.

Reply:
Hm... I'm torn if we actually should fail the pipeline if the codecov report fails to upload. I guess the built binary could be fine. I just don't want to get into a situation where the published binary doesn't have an associated codecov. 

I think the codecov bot will block the PR if codecov is too low

Reply:
@florkbr This shell script (and hopefully any other scripts too) uses the so-called [Bash strict mode](http://redsymbol.net/articles/unofficial-bash-strict-mode/)

```bash
set -euo pipefail
```

> The `set -e` option instructs bash to immediately exit if any command has a non-zero exit status.

So if the above `curl` command (or any other command) fails with non-zero exit code, this script as a whole will fail with that exit code. IMO this is a very sensible default for user scripts.

> Hm... I'm torn if we actually should fail the pipeline if the codecov report fails to upload.

We may treat the commands inside the if-block `[[ "${JOB_TYPE}" != "local" ]]` as ones that could potentially fail without this script failing as a result. I'd do this later as a follow-up change.